architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+5. Dashboard is not public facing.
+6. Access to Dashboard is using HTTPS.
+7. Authentication mechanism provided by Backstage is robust.
+
+## Data Flow
+
+### Diagram
+
+![Radius Dashboard](2024-08-dashboard-component-threat-model/dashboard-tm.png)
+
+1. User types the backstage url and accesses Radius plugin
+2. Request reaches the dashboard pod in `radius-system` namespace in kubernetes cluster.
+3. The dashboard service sends a Radius API request to UCP.
+4. UCP works with ApplicationsCore-RP and sends response back to Dashboard SPA.
+5. Dashboard SPA contructs the visuals using components from backstage core, rad-component and data in API response and responds with appropriate page to the user. 
+
+### Threats
+ 
+#### Threat 1: DoS
+
+**Description**
+
+A client can access Dashboard repeatedly or fetch the page in a loop.
+
+**Impact**
+
+Due to the volume of requests Dashboard as well as the UCP, AppCore-RP components involved in serving the request could run out of resource to serve a legitimate request.
+   
+**Mitigation**:
+
+1. Access to Dashboard should be provided to trusted users.  The [Backstage  permissions system](https://backstage.io/docs/permissions/overview) should be enabled and configured to restrict access as necessary.
+   
+**Status**:
+
+Active. Operators are expected to configure the system and limit access to Dashboard portal. 
+
+2. Audit logs should be enabled to monitor and report on suspicious user activity. 
+
+**Status**
+
+[In Progress](https://github.com/backstage/backstage/issues/23950)
+
+#### Threat 2: Information Disclosure by unauthorized access to application information
+
+**Description**
+
+Access to app graph through the portal can provide information on dependency between app components.
+
+**Impact**:
+
+A malicious user can utilize the app graph to stage effective attack by targeting a component that has most dependency.
+
+**Mitigation**:
+
+1. Access to Dashboard portal should be provided to trusted users. While we dont expose any secrets in db, users should still enable authentication and secure access to data based on roles.
+   
+**Status**:
+
+Active. Operators are expected to configure the [Backstage  permissions system](https://backstage.io/docs/permissions/overview) to restrict access as necessary. 
+
+2. Audit logs should be enabled to monitor and report on suspicious user activity. 
+
+**Status**
+
+[In Progress](https://github.com/backstage/backstage/issues/23950)
+
+#### Threat 3: Spoofing dashboard service-account can cause DoS 
+**Description**
+
+If an unauthorized user or malicious actor gtampers with cluster,
+he can create services with dashboard service-account to send Radius API requests. 
+
+**Impact**:
+
+A malicious user can utilize Radius API and create too many resources. 
+
+This is because dashboard service-account is allowed get, list, and post requests to UCP
```

Comment:
Would be helpful to set some context in the architecture section on why `post` permission is needed.